src/picongpu/include/particles/manipulators/FreeImpl.hpp

```diff
@@ -31,35 +32,138 @@ namespace particles
+         *
+         * @tparam DeferFunctor is used to defer the functor type evaluation to enable/disable
+         *                       the constructor
+         * @param currentStep current simulation time step
+         * @param is used to enable/disable the constructor (do not pass any value to this parameter)
+         */
+        template< typename DeferFunctor = Functor >
+        HINLINE FreeImpl(
+            uint32_t currentStep,
+            typename std::enable_if<
+                std::is_constructible<
+                    DeferFunctor,
+                    uint32_t
+                >::value
+            >::type* = 0
+        ) : Functor( currentStep )
+        {
+        }
+
+        /** constructor
+         *
+         * is activated if the user functor has a default constructor
+         *
+         * @tparam DeferFunctor is used to defer the functor type evaluation to enable/disable
+         *                       the constructor
+         * @param current simulation time step
+         * @param is used to enable/disable the constructor (do not pass any value to this parameter)
+         */
+        template< typename DeferFunctor = Functor >
+        HINLINE FreeImpl(
+            uint32_t,
+            typename std::enable_if<
+                std::is_constructible< DeferFunctor >::value
+            >::type* = 0
+        ) : Functor( )
+        {
+        }
+
+        /** call user functor
+         *
+         * calls the user functor if both given particles are valid
+         *
+         * @param cell index within the local volume
+         * @param particleSpecies1 first particle
+         * @param particleSpecies2 second particle, can be equal to the first particle
+         * @param isParticle1 define if the reference @p particleSpecies1 is valid
+         * @param isParticle1 define if the reference @p particleSpecies2 is valid
+         * @return void is used to enable the operator if the user functor except two arguments
+         */
+        template<
+            typename T_Particle1,
+            typename T_Particle2
+        >
+        DINLINE auto operator()(
+            DataSpace<simDim> const &,
+            T_Particle1 & particleSpecies1,
+            T_Particle2 & particleSpecies2,
+            bool const isParticle1,
+            bool const isParticle2
+        )
+        -> decltype(
+            std::declval< Functor >()(
+                particleSpecies1,
+                particleSpecies2
+            )
+        )
         {
-            Functor userFunctor;
-            userFunctor(particleSpecies1, particleSpecies2);
+            if( isParticle1 && isParticle2 )
+            {
+                Functor::operator()(
+                    particleSpecies1,
+                    particleSpecies2
+                );
+            }
         }
-    }
 
-};
+        /** call user functor
+         *
+         * calls the user functor if the first particle is valid
```

Comment:
`if the first particle is valid` is not entirely true in that context. it's about the user functor and if it operates on two particles (binary) or one (unary).